promotion/controllers.go

```diff
@@ -233,3 +234,31 @@ func GetClaim(service *Service) handlers.AppHandler {
 
 	})
 }
+
+// AggregateClaims returns an aggregation of grants claimed by a given wallet
+func AggregateClaims(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		claimType := chi.URLParam(r, "claimType")
+		walletID := r.URL.Query().Get("walletID")
+
+		if len(walletID) == 0 || !govalidator.IsUUIDv4(walletID) {
+			return handlers.WrapError(errors.New("walletID is not a uuid"), "", http.StatusBadRequest)
```

Comment:
as mentioned I'd prefer if we returned an `AppError` directly here that parallels the structure we get out of other validation errors